core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/CollectInformation.scala

```diff
@@ -81,11 +81,53 @@ class CollectInformation(apps: Seq[ApplicationInfo]) extends Logging {
 
   // get read data schema information
   def getDataSourceInfo: Seq[DataSourceProfileResult] = {
-    val filtered = apps.filter(_.dataSourceInfo.size > 0)
-    val allRows = filtered.flatMap { app =>
+    val dataSourceApps = apps.filter(_.dataSourceInfo.size > 0)
+    val sqlAccums = CollectInformation.generateSQLAccums(dataSourceApps)
+
+    // Metrics to capture from event log to the result
+    val buffer_time: String = "buffer time"
+    val scan_time = "scan time"
+    val data_size = "size of files read"
+    val decode_time = "GPU decode time"
+
+    // This is to save the metrics which will be extracted while creating the result.
+    case class IoMetrics(
+        var buffer_time: String,
+        var scan_time: String,
+        var data_size: String,
+        var decode_time: String)
+
+    def getIoMetrics(sqlAccums: Seq[SQLAccumProfileResults]): IoMetrics = {
+      val finalRes = IoMetrics("", "", "", "")
+      sqlAccums.map(accum => accum.name match {
+        case `buffer_time` => finalRes.buffer_time = accum.max_value.toString
+        case `scan_time` => finalRes.scan_time = accum.max_value.toString
+        case `data_size` => finalRes.data_size = accum.max_value.toString
+        case `decode_time` => finalRes.decode_time = accum.max_value.toString
+      })
+      finalRes
+    }
+
+    val allRows = dataSourceApps.flatMap { app =>
+      val appSqlAccums = sqlAccums.filter(sqlAccum => sqlAccum.appIndex == app.index)
+
+      // Filter appSqlAccums to get only required metrics
+      val dataSourceMetrics = appSqlAccums.filter(sqlAccum => sqlAccum.name.contains(buffer_time)
+        || sqlAccum.name.contains(scan_time) || sqlAccum.name.contains(decode_time)
+        || sqlAccum.name.equals(data_size))
+
       app.dataSourceInfo.map { ds =>
-        DataSourceProfileResult(app.index, ds.sqlID, ds.format, ds.location,
-          ds.pushedFilters, ds.schema)
+        val sqlIdtoDs = dataSourceMetrics.filter(
+          sqlAccum => sqlAccum.sqlID == ds.sqlID && sqlAccum.nodeID == ds.nodeId)
+        if (!sqlIdtoDs.isEmpty) {
+          val ioMetrics = getIoMetrics(sqlIdtoDs)
+          DataSourceProfileResult(app.index, app.gpuMode, ds.sqlID, ds.nodeId,
+            ds.format, ioMetrics.buffer_time, ioMetrics.scan_time, ioMetrics.data_size,
+            ioMetrics.decode_time, ds.location, ds.pushedFilters, ds.schema)
+        } else {
+          DataSourceProfileResult(app.index, app.gpuMode, ds.sqlID, ds.nodeId,
+            ds.format, "NA", "NA", "NA", "NA", ds.location, ds.pushedFilters, ds.schema)
```

Comment:
this is inconsistent with above line 101 where we initialize to  empty string.  I would change these just to be empty string.  Or really these are meant to be numbers so initialize them to 0. I get that we are trying to indicate that it didn't have any of these but if we are trying to aggregate after the fact having everything the same type would be better.